src/resizing.jl

```diff
@@ -156,34 +156,58 @@ end
 # imresize
 imresize(original::AbstractArray, dim1, dimN...) = imresize(original, (dim1,dimN...))
 
-function imresize{T,N}(original::AbstractArray{T,N}, short_size::NTuple)
+function imresize(original::AbstractArray, short_size::Tuple)
     len_short = length(short_size)
-    new_size = ntuple(i -> (i > len_short ? size(original,i) : short_size[i]), N)
+    new_size = ntuple(i -> (i > len_short ? odims(original, i, short_size) : short_size[i]), N)
```

Comment:
Here's an illustrative example:
```julia
julia> foo() = n > 2
foo (generic function with 1 method)

julia> @code_warntype foo()
Variables:
  #self#::#foo

Body:
  begin 
      SSAValue(0) = Main.n
      return (2 < SSAValue(0))::Any
  end::Any
```
You can see that the inferred code will try to look up a variable `Main.n` when this code runs. If you look at this with `@code_llvm`, you'll see a block
```julia
notfound:                                         ; preds = %top
  %11 = call %jl_value_t* @jl_get_binding_or_error(%jl_value_t* inttoptr (i64 139624985559056 to %jl_value_t*), %jl_value_t* inttoptr (i64 139624983216904 to %jl_value_t*))
  store %jl_value_t* %11, %jl_value_t** @delayedvar70580, align 8
  br label %found
```
which, from the names alone, probably convinces you that the *compiled* code includes a *runtime* variable lookup. In other words, it compiles the lookup into the code; that's why it's a runtime failure.